apiserver/facades/agent/machine/machiner.go

```diff
@@ -153,17 +173,40 @@ func (api *MachinerAPI) RecordAgentStartTime(args params.Entities) (params.Error
 // MachinerAPIV1 implements the V1 API used by the machiner worker. Compared to
 // V2, it lacks the RecordAgentStartTime method.
 type MachinerAPIV1 struct {
+	*MachinerAPIV2
+}
+
+// MachinerAPIV2 implements the V1 API used by the machiner worker. Compared to
+// V2, it backfills the missing origin in NetworkConfig.
+type MachinerAPIV2 struct {
 	*MachinerAPI
 }
 
 // NewMachinerAPIV1 creates a new instance of the V1 Machiner API.
 func NewMachinerAPIV1(st *state.State, resources facade.Resources, authorizer facade.Authorizer) (*MachinerAPIV1, error) {
-	api, err := NewMachinerAPI(st, resources, authorizer)
+	api, err := NewMachinerAPI(st, resources, authorizer, func(st *state.State, fn common.GetAuthFunc) NetworkConfigAPI {
+		return networkingcommon.NewNetworkConfigAPIV1(st, fn)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO (stickupkid): I'm not a fan of this, but I don't see a way to
```

Comment:
How about this? 

Instead of versioning APIs for _Machiner_ and _NetworkConfig_, version _Machiner_ and _Provisioner_, so the backfill goes in the parent facades.

You could also make the backfill a method of `params.SetMachineNetworkConfig` itself so that you don't duplicate the logic. So V2 just calls the backfill before calling the V1 method...